src/main/java/com/gotocompany/depot/maxcompute/MaxComputeSinkFactory.java

```diff
@@ -67,10 +67,12 @@ public Sink create() {
                 messageParser,
                 partitioningStrategy,
                 maxComputeSinkConfig,
-                sinkConfig
+                sinkConfig,
+                new Instrumentation(statsDReporter, RecordDecorator.class),
```

Comment:
pass the StatsDReporter and let the concerned class instantiate the instrumentation